mp2-v1/src/api.rs

```diff
@@ -480,7 +496,74 @@ pub fn no_provable_metadata_hash<I: IntoIterator<Item = ColumnID>>(column_ids: I
         .collect_vec();
     let digest = map_to_curve_point(&inputs);
 
-    combine_digest_and_block(digest)
+    let metadata_hash = combine_digest_and_block(digest);
+    if provable_data_commitment {
+        // add the data commitment prefix to the metadata hash, to keep track
+        // of whether a commitment to the data of the table is used as root of trust
+        add_provable_data_commitment_prefix(metadata_hash)
+    } else {
+        metadata_hash
+    }
+}
+
+pub struct TableRow<PrimaryIndex> {
+    primary_index: PrimaryIndex,
+    row_values: CellCollection<PrimaryIndex>,
+}
```

Comment:
Is there a way to for the caller NOT to have to construct this manually ? (which is now a struct whose single purpose as far as i can see is to be created to be passed to a method) Ideally he could use directly the format of the data he submits OR something that ryhope returns directly with a single call. I dont think it's neither in this case.

Reply:
This API is mostly meant to be employed by external users to re-compute the commitment over their data, so my idea is that this would be wrapped by some other API on DQ which is compatible with the format of the data used for off-chain tables. But the code here I think should be independent from this data format, as it could be freely changed in DQ. Maybe I can try to use only a slice of `CellCollection`, the code on the integration test will be a bit more confusing since I'll need to modify the cells fetched from ryhope to have the correct primary index, but the interface for this API will become a bit cleaner. Wdyt?

Reply:
Actually while reasoning on the changes I think it will be more convoluted to use `CellCollection` as input for the entire row, since the primary index will need to be copied in all the cells. So I would either leave it like this or replace `CellCollection` with another data structure containing only column identifiers and corresponding values. Wdyt?